routes/helpers/localize_results.js

```diff
@@ -0,0 +1,37 @@
+const localize =  (locale, dbResultsObj) => {
+
+  // keynames in DB that contain translatable content follow pattern 'name___locale'
+  const separator = '___';
+  const translatableKeyReg = /\_{3}.+$/;
+
+  // gets array of translatable values from DB results
+  const translatableFields = Object.keys(dbResultsObj)
+    .filter(key => translatableKeyReg.test(key))
+    .map(filteredKey => filteredKey.replace(translatableKeyReg, ''));
+
+  // create new results object with relevant data only
+  return Object.keys(dbResultsObj)
+    .filter(key => {
+      const [ type, loc, ] = key.split(separator);
+      return translatableFields.indexOf(type) < 0 || loc === locale;
+    }) // change key name so that '___LOCALE' part is discarded
+    .reduce((a, b) =>
+      Object.assign({}, a, { [b.replace(translatableKeyReg, '')] : dbResultsObj[b], }), {});
+};
+
+const localizeResults = (locale, dbResults) => {
+
+  // if currentLocale === defaultLocale, do nothing
```

Comment:
I think you should get rid of the idea of a default local.

In one part of you application, you should set a default, then, everywhere else, assume it is set?

Thoughts?